pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) error {
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
+	}
+
+	spBytes, err := json.Marshal(servicePrincipal)
+	if err != nil {
+		return err
+	}
+
+	// TODO: What if a customer adds/modifies a pullsecret after the install ?
+	// TODO: All parameter locations to be verified
+	m.hiveWorkloadCluster = &hive.WorkloadCluster{
+		Subscription:                m.subscriptionDoc.ID,
+		Name:                        m.doc.OpenShiftCluster.Name,
+		Location:                    m.doc.OpenShiftCluster.Location,
+		BaseDomain:                  m.env.Domain(),
+		BaseDomainResourceGroupName: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID,
```

Comment:
`*.aroapp.io` is going to be a base domain. And in prod we do not put child DNS zones into cluster resource groups. [This](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/14008788#16266901) might be useful.